packages/blocks/src/api/raw-handling/nested-list-converted.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Internal dependencies
+ */
+
+function isList( node ) {
+	return node.nodeName === 'OL' || node.nodeName === 'UL';
+}
+
+function createLineBreak( doc ) {
+	return doc.createElement( 'br' );
+}
+
+function createBullet( doc, child, isNested ) {
+	const node = doc.createElement( 'pre' );
+	const isOrdered = child.parentElement.nodeName === 'OL';
+	let bullet = '- ';
+
+	if ( isOrdered ) {
+		const index = Array.from( child.parentElement.childNodes )
+			.filter( ( item ) => item.nodeName === 'LI' )
+			.indexOf( child );
+		bullet = `${ index + 1 }. `;
+	}
+	if ( isNested ) {
+		bullet = `  ${ bullet }`;
```

Comment:
@ellatrix I made some progress on this. Now that the transformation happens during an earlier stage, the indentation whitespace I am adding gets stripped away. Any ideas on how to preserve it so nested lists look correct?

Reply:
Currently this:

![Screenshot 2566-10-26 at 17 09 11](https://github.com/WordPress/gutenberg/assets/6851384/3afaf01f-58e7-4e98-a133-1bdbacc3b35b)

Gets rendered as this:

<img width="706" alt="Screenshot 2566-10-26 at 17 09 31" src="https://github.com/WordPress/gutenberg/assets/6851384/edb5bdda-4043-4947-9b2b-d07aac797685">

Reply:
The whitespace is being stripped out by `htmlFormattingRemover` in the same file. What are the implications of changing the code so it doesn't do this for `td`? 

Or maybe the transformation I am adding can add an attribute `data-preserve-whitespace="true"` which `htmlFormattingRemover` can check? (it can also remove the attribute after reading it)

Reply:
I've been trying to use an attribute like `data-preserve-whitespace` or just a flag like `node.preserveWhiteSpace` but these don't work. Any attributes need to be on the schema(s), or they get removed. And adding new properties to nodes like `node.preserveWhiteSpace` does not work because these properties get removed during other transformations.

Reply:
One other thing I just tried was to record the XPath of the elements where you want to preserve white space, e.g. `body/b/div/table/tbody/tr[3]/td` and then check these paths in `htmlFormattingRemover` and return early. However, this doesn't work because in `htmlFormattingRemover`, the same path now looks like `body/table/tbody/tr[3]/td` due to other transform functions removing nodes. This is harder than I expected!

Reply:
There's no point in preserving the whitespace since HTML rendering collapses that anyway: https://codepen.io/iseulde/pen/KKJNPyN

You'll probably need to use either non breaking spaces or just omit list indentation entirely.

Reply:
@ellatrix preserving whitespace is fundamental to what we're trying to do as we want to preserve the structure of the list, and lists can be nested. I had already tried inserting a `span` with content `\u00A0`, but it got by a later transform. I have found another potential solution and left a comment on that line.

Reply:
Tbh, I'm not even sure if we should add a bunch of non breaking spaces. It feels wrong. It's so easy for these to get copied to other places and people won't know why lines don't break.

I'd prefer I we just collapse all the indentation. Anything wrong with that? We shouldn't pretend this is a real list.

Reply:
I feel whitespace is essential for the original use case which is to allow to pasting nested tables in from Google Docs. However, it does feel a bit specific and is probably not appropriate for all paste handling, which is how it is being applied. We could make it work for only table cells that came from Google Docs, but that feels a bit random. 

@danielbachhuber do you have any input here? You could just create images of you tables and use them instead 😁.

Reply:
I'd be fine to remove the indentation whitespace as long as the newlines are preserved. But, as a user, I'd expect the indentation whitespace to persist. I'd suggest seeing how Dropbox Paper, Microsoft Word, and Google Docs handle it, and see if there's some obvious answer from those.

Reply:
Good suggestion. I tried Dropbox and Word - they both support tables. Word collapses the nested lists though on paste even though it allows me to nest the items manually. Dropbox adds a second column so is ignoring the `colspan`. 

<img width="809" alt="Screenshot 2566-11-09 at 12 09 31" src="https://github.com/WordPress/gutenberg/assets/6851384/7ef1556e-e4b8-4de6-a853-c873a684f1d4">

![image](https://github.com/WordPress/gutenberg/assets/6851384/fef7cd68-61c3-49a8-938c-dfefee5e9ac8)

>I'd be fine to remove the indentation whitespace as long as the newlines are preserved. 

I think this is a good point too. There is indentation in the source so you'd want it in the destination too. So maybe it not such a random feature.

Reply:
But what are they using to represent the tables? Are those just unicode bullets (text) or is it actual list markup? Maybe test in a couple of more places?
* Apple Notes (should be identical to Pages)
* Anything else you use?
* Could you create a table with a list in HTML and paste it in Google Docs?

Reply:
>I don't want (1) these "fake" lists to be too good that it becomes confusing to understand what you're dealing with and (2) people to thing you can indent with spaces, because you cannot

@ellatrix, this feels like a bug to me. It's supposed to be a _WYSIWYG_ editor. I tried creating the same "fake" list in the editor and it's fine but on the frontend it is collapsed. 

<img width="903" alt="Screenshot 2566-11-09 at 15 11 56" src="https://github.com/WordPress/gutenberg/assets/6851384/42e636ac-64aa-4d5c-a53a-80065c9f812c">

<img width="723" alt="Screenshot 2566-11-09 at 15 13 13" src="https://github.com/WordPress/gutenberg/assets/6851384/36c0e55a-f6bb-472d-9820-ecd0f07ac4b9">

Reply:
>or we need to alternate between inserting spaces and non breaking spaces.

Can confirm this works. Can we go with that?

Reply:
I just pushed my latest code.